code/datums/components/crafting/_recipes.dm

```diff
@@ -31,6 +31,16 @@
 	var/list/structures
 	///Should only one object exist on the same turf?
 	var/one_per_turf = FALSE
+	/// Setting this to true will effectively set check_direction to true.
+	var/is_fulltile = FALSE
+	/// If this atom should run the direction check, for use when building things like directional windows where you can have more than one per turf
+	var/check_direction = FALSE
+	/// If the atom requires a floor below
+	var/on_solid_ground = FALSE
+	/// If the atom checks that there are objects with density in the same turf when being built. TRUE by default
+	var/check_density = TRUE
+	/// Bitflag of additional placement checks required to place. (STACK_CHECK_CARDINALS|STACK_CHECK_ADJACENT|STACK_CHECK_TRAM_FORBIDDEN|STACK_CHECK_TRAM_EXCLUSIVE)
+	var/placement_checks = NONE
```

Comment:
at this point all these flags should be... flags tbh.

Reply:
😅

Reply:
we live in a bool society